tests/integration/rules/numeric.phpt

```diff
@@ -0,0 +1,51 @@
+--FILE--
+<?php
+require 'vendor/autoload.php';
+
+use Respect\Validation\Exceptions\AllOfException;
+use Respect\Validation\Exceptions\NumericValException;
+use Respect\Validation\Validator as v;
+
+try {
+    v::numericVal()->check('a');
+} catch (NumericValException $e) {
+    echo $e->getMessage().PHP_EOL;
+}
+
+try {
+    v::numericVal()->assert('a');
+} catch (AllOfException $e) {
+    echo $e->getFullMessage().PHP_EOL;
+}
+
+try {
+    v::numericVal()->setName('Field')->check(null);
+} catch (NumericValException $e) {
+    echo $e->getMessage().PHP_EOL;
+}
+
+try {
+    v::numericVal()->setName('Field')->assert('');
+} catch (AllOfException $e) {
+    echo $e->getFullMessage().PHP_EOL;
+}
+
+try {
+    v::not(v::numericVal())->check('1');
+} catch (NumericValException $e) {
+    echo $e->getMessage().PHP_EOL;
+}
+
+try {
+    v::not(v::numericVal())->assert('1');
+} catch (AllOfException $e) {
+    echo $e->getFullMessage().PHP_EOL;
+}
+?>
+--EXPECTF--
+"a" must be numeric
+- "a" must be numeric
```

Comment:
The newer tests are testing:

1. `v::rule()->check($input)`
2. `v::not(v::rule())->check($input)`
3. `v::rule()->assert($input)`
4. `v::not(v::rule())->assert($input)`

These 4 in that order should be enough.